### PR TITLE
Deprecate Array.sorted in favor of using Sort.sorted

### PR DESCRIPTION
**Pull request by stonea**
This is a followup item for the Array module review.

In the interest of keeping the interface to array streamlined we decided to remove a small handful of procedures.

One of these is Array.sorted(), which returns an iterator over sorted array elements. There is already an existing sorted iterator in Sort (in fact in most cases Array.sorted() ends up just calling out to it). Array.sorted() had some additional functionality to handle associative arrays so I updated Sort.sorted to handle them as well.

I've verified that with these changes paratest passes.

This is a part of four similar PRs: 
* https://github.com/chapel-lang/chapel/pull/20025
* https://github.com/chapel-lang/chapel/pull/20062
* https://github.com/chapel-lang/chapel/pull/20077
* https://github.com/chapel-lang/chapel/pull/20078

### CODE REVIEW

File: modules/packages/Sort.chpl [hunk 1]
```diff
@@ -550,13 +556,22 @@ proc isSorted(Data: [?Dom] ?eltType, comparator:?rec=defaultComparator)
...
-
-  sort(y, comparator=comparator);
-  for i in y do
-    yield i;
+  if isArrayValue(x) && Reflection.canResolveMethod(x._value, "dsiSorted", comparator)
+  {
+    for i in x._value.dsiSorted(comparator) {
+      yield i;
+    }
+    return;
```
Comment (jeremiah-corrado) [inline modules/packages/Sort.chpl:564]: Do we also want to include this check (without the comparator):
```chapel
} else if Reflection.canResolveMethod(_value, "dsiSorted") {
        compilerError(_value.type:string + " does not support dsiSorted(comparator)");
}
```
like the old ChapelArray implementation?

**Review by jeremiah-corrado — COMMENTED** [review]
My brain no longer recognizes "sorted" as a word, but I think this looks good overall!

I just had one suggestion on the deprecation message, and a question about the additions you made to Sort.sorted() for associative arrays.

File: modules/internal/ChapelArray.chpl [hunk 2]
```diff
@@ -1590,6 +1590,7 @@ module ChapelArray {
     }
 
     /* Yield the array elements in sorted order. */
+    deprecated "sorted is deprecated"
```
[the remark this answers is not in the document]
Comment (stonea) [reply modules/internal/ChapelArray.chpl]: Updated to `deprecated "Array.sorted is deprecated, use 'Sort.sorted' instead"`

**Review by jeremiah-corrado — APPROVED** [review]

**Comment by stonea** [comment]
Closing this in favor of merging https://github.com/chapel-lang/chapel/pull/20140 (which combines this PR with related ones. I'm combining them ahead of time since there's some overlap in modified tests.)

**Pull request closed without merging**